### PR TITLE
[Validator]: Fix append propertyPath in collection

### PR DESCRIPTION
**Pull request by jbdelhommeau**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #22298
| License       | MIT

I raised a bug in validator component.
I try to build violation for a children in my collection and affect a custom path to target my specific child.

However, in this context the propertyPath inherit of basePath sufixed with '.data'. And this breaks the message of validation.

Exemple:

BasePath of violation path is :  ``` children[customers].data ```

AtPath: ```children[1].children[followupMailingList].data```

Final Violation path is: ``` children[customers].data.children[1].children[followupMailingList].data ```

The right final violation path sould be (without the .data of basePath):  ``` children[customers].children[1].children[followupMailingList].data ```

https://github.com/symfony/symfony/issues/22298

### CODE REVIEW

File: src/Symfony/Component/Validator/Util/PropertyPath.php [hunk 1]
```diff
@@ -41,6 +41,10 @@ public static function append($basePath, $subPath)
                 return $basePath.$subPath;
             }
 
+            if (preg_match('/(children.*)(.data)$/', $basePath, $match)) {
```
Comment (curry684) [inline src/Symfony/Component/Validator/Util/PropertyPath.php]: Are you aware that this regexp happily matches `mychildrenlovebigdata`?

File: src/Symfony/Component/Validator/Util/PropertyPath.php [hunk 2]
```diff
@@ -41,6 +41,10 @@ public static function append($basePath, $subPath)
                 return $basePath.$subPath;
             }
 
+            if (preg_match('/^(children\[.*\])(\.data)$/', $basePath, $match)) {
```
Comment (jbdelhommeau) [inline src/Symfony/Component/Validator/Util/PropertyPath.php]: Review regex: 

https://regex101.com/r/TEMcRi/1/

File: src/Symfony/Component/Validator/Util/PropertyPath.php (hunk 1 above)
Comment (jbdelhommeau) [inline src/Symfony/Component/Validator/Util/PropertyPath.php]: Thank you I review regex ;)

**Comment by Nek-** [comment]
Hi !

Thanks for your contribution :) . Could you add a unit test to this PR ?

**Comment by jbdelhommeau** [comment]
Ok I add unit tests and switch branch to 2.8. But my source branch is always master branch of symonfy. Is it a problem ? Now all checks have failed :(

**Comment by fabpot** [comment]
Something went wrong with this PR. @jbdelhommeau Can you open a new one? Thanks.

**Comment by jbdelhommeau** [comment]
Ok i created a new pull request targeted on branch 2.8

https://github.com/symfony/symfony/pull/22609

**Pull request closed without merging**